Main/include/ItemSelectionWheel.hpp

```diff
@@ -251,8 +269,26 @@ class ItemSelectionWheel
 	{
 		if (m_SourceCollection().empty())
 			return;
-		uint32 selection = Random::IntRange(0, (int32)m_sortVec.size() - 1);
-		SelectItemBySortIndex(selection);
+
+		// If the randomized vector of charts has not been initialized or has
+		// been exhausted, we regenerate it from the current m_sortVec
+		if (m_randomVec.empty()) {
+			m_randomVec = m_sortVec;
```

Comment:
Would it make more sense to keep track of the charts that have already been played rather than creating a o(n) vector the first time a random chart is played?


I'm not sure how bg a deal since it is just a vector of integers, but something to consider

Reply:
I definitely agree with you. At first I thought that most folders would be relatively small so O(n) would be pretty inconsequential, but nautica has the potential to cause some large folder sizes. I have updated the algorithm to address this.